plugins/module_utils/session_utils.py

```diff
@@ -242,25 +245,29 @@ def _url_common_args_spec(self, method, api_timeout, headers=None):
                 - follow_redirects (str): The policy for following redirects.
 
         """
-        req_header = self._headers
-        if headers:
-            req_header.update(headers)
         if api_timeout is None:
             api_timeout = self.timeout
         if self.ca_path is None:
             self.ca_path = self._get_omam_ca_env()
-        url_kwargs = {
+        req_header = self._headers
+        if headers:
+            req_header.update(headers)
+        url_params = {
             "method": method,
             "validate_certs": self.validate_certs,
             "ca_path": self.ca_path,
             "use_proxy": self.use_proxy,
             "headers": req_header,
             "timeout": api_timeout,
-            "follow_redirects": 'all',
+            "follow_redirects": 'all'
         }
-        return url_kwargs
+        if url_kwargs is None:
+            return url_params
+        else:
+            url_kwargs.update(url_params)
```

Comment:
Can we keep this simple.
if url_kwargs:
    url_params.update(url_kwargs)
return url_params